simpleviewinc/keg-hub
# Zen 581 svg props

**Ticket**: [ZEN-581](https://jira.simpleviewtools.com/browse/ZEN-581)

> This PR is branched off of [this keg-components PR](https://github.com/simpleviewinc/keg-hub/pull/154)
>   * Please go do that PR before this one
> This PR should be tested in conjunction with [this events-force PR]()

## Context
* The SVG Icon component can not accept the `className` prop
* The UK team needs a `className` on the icons


## Goal
* Allow the SVG Icon to accept a `className` prop
* Added example for using the SVGIcon component

### Updates

* `repos/keg-components/configs/aliases.json`
  * Added alias for SvgIcon component
* `repos/keg-components/src/components/loading/loading.js`
  * Add default class name for loading indicator
* `repos/keg-components/src/components/svgIcon/*`
  * Added story-book examples for SvgIcon component
  * Updated SvgIcon component to follow the same pattern as other components
  * Added helper hook for adding the `className` to the component


### Tests

**Setup**
* If [this PR](https://github.com/simpleviewinc/keg-cli/pull/64) has not yet been merged
  * Switch to that branch for the `keg-cli` => `keg cli && keg pr 64`
  * This ensures `keg-components` is treated as a `linked tap`, and not an `internal app`
  * You may also have to link `keg-components` as a tap
    * Run `keg && cd repos/keg-components && keg tap link components`

**Test 1**
* Run this package => `keg components pack run keg-components:zen-581-svg-props`
  * Navigate to http://components-zen-581-svg-props.local.keghub.io/ in your browser
  * Navigate to `Components => Display => SvgIcon`
    * Make sure you select the `docs` tab
    * [Looks like this](https://prnt.sc/127poc1)
* Ensure the component is displaying as expected

**Test 2**
* Open the inspector and validate that the custom className `keg-animated-icon` exists on the svg element
* [Looks like this](https://prnt.sc/127ptg2)
